tests/json/test_load_from_json.py

```diff
@@ -311,48 +408,64 @@ def test_GIVEN_component_with_nx_class_WHEN_loading_from_json_THEN_new_model_con
     assert json_reader.entry.instrument.component_list[1].nx_class == component_class
 
 
-def test_GIVEN_transformation_with_matching_name_WHEN_finding_transformation_by_name_THEN_transformation_is_returned(
-    json_reader, component_with_transformation
+def test_GIVEN_json_with_component_depending_on_transfrom_WHEN_loaded_THEN_component_in_model_contains_transform(
+    json_dict_with_component_and_transform, json_reader
 ):
-    transformation = component_with_transformation.transforms[0]
-    assert transformation == json_reader._get_transformation_by_name(
-        component_with_transformation, transformation.name, "DependentComponentName"
-    )
+    json_reader._load_from_json_dict(json_dict_with_component_and_transform)
+    for component in json_reader.entry.instrument.component_list:
+        if component.name == "test_component":
```

Comment:
This unit test is too likely to "fail silently". You need to make sure (guarantee) that the asserts are reached or fail the test if they are not.

Reply:
Fixed, thanks for pointing out